### PR TITLE
Verify end of tour date not in past

### PR DESCRIPTION
**Pull request by evdhoudt**
Closes NCI-Agency/anet#1367

### CODE REVIEW

File: client/src/models/Person.js [hunk 1]
```diff
@@ -128,6 +129,17 @@ export default class Person extends Model {
               )
             : schema.nullable()
         )
+        .when(["role", "status", "uuid"], (role, status, uuid, schema) =>
+          Person.isAdvisor({ role }) && Person.isNewFormUser({ status, uuid })
+            ? schema.test(
+                "end-of-tour-date",
+                `The ${
+                  Settings.fields.person.endOfTourDate
+                } date must be set in the future`,
```
Comment (gjvoosten) [inline client/src/models/Person.js]: Change *must be set in the future* to *must be in the future*.

File: client/src/models/Person.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ export default class Person extends Model {
   static yupSchema = yup
     .object()
     .shape({
+      uuid: yup.string().default(""),
```
Comment (gjvoosten) [inline client/src/models/Person.js]: Use `yup.string().nullable().default(null)` (a new object should have a null uuid, not the empty string).

File: client/src/models/Person.js [hunk 3]
```diff
@@ -128,6 +129,17 @@ export default class Person extends Model {
               )
             : schema.nullable()
         )
+        .when(["role", "status", "uuid"], (role, status, uuid, schema) =>
```
Comment (gjvoosten) [inline client/src/models/Person.js]: Why not combine this with the `.when` directly above it?

**Comment by evdhoudt** [comment]
> Also run `yarn run lint-fix` before pushing to make sure code is properly formatted.

Hmm... does this mean that I cannot use vscode for formatting? Both prettier and eslint reformats the file in other ways than our linter.

File: client/src/models/Person.js [hunk 4]
```diff
@@ -34,6 +34,7 @@ export default class Person extends Model {
   static yupSchema = yup
     .object()
     .shape({
+      uuid: yup.string().nullable().default(null),
```
Comment (gjvoosten) [inline client/src/models/Person.js]: `yarn run lint-fix`

File: client/src/models/Person.js [hunk 5]
```diff
@@ -119,15 +120,24 @@ export default class Person extends Model {
         .label(Settings.fields.person.phoneNumber),
       endOfTourDate: yupDate
         .nullable()
-        .when("role", (role, schema) =>
+        .when(["role", "status", "uuid"], (role, status, uuid, schema) => {
```
Comment (gjvoosten) [inline client/src/models/Person.js]: You're not returning anything (should return the schema). Conditions can be simplified, I'll have a go at it.

**Comment by evdhoudt** [comment]
Do we need a second opinion or can this PR be merged?

**Comment by VassilIordanov** [comment]
> Do we need a second opinion or can this PR be merged?

My bad, apologies for the delay.. PR is good but the requirement was not fully flushed. I've added this to the requirement:
https://github.com/NCI-Agency/anet/issues/1367#issuecomment-489062725

**Comment by evdhoudt** [comment]
@VassilIordanov @gjvoosten I've fixed the EOT warning.

It only shows when creating a new user during the on-boarding proces.
Otherwise a warning is shown when editing or creating a person with EOT in the past.

Also added wdio tests to check the validation during on-boarding.

**Pull request merged**